pkg/util/limiter/query_limiter.go

```diff
@@ -77,3 +85,14 @@ func (ql *QueryLimiter) uniqueSeriesCount() int {
 	defer ql.uniqueSeriesMx.Unlock()
 	return len(ql.uniqueSeries)
 }
+
+// AddChunkBytes adds the input chunk size in bytes and returns an error if the limit is reached.
+func (ql *QueryLimiter) AddChunkBytes(chunkSizeInBytes int) error {
+	if ql.maxChunkBytesPerQuery == 0 {
+		return nil
+	}
+	if ql.chunkBytesCount.Add(int64(chunkSizeInBytes)) > int64(ql.maxChunkBytesPerQuery) {
+		return validation.LimitError(fmt.Sprintf(ErrMaxChunkBytesHit, ql.maxChunkBytesPerQuery))
```

Comment:
Nit: Same comment as in `AddSeries` -- no need to return `validation.LimitError` from here. Simple  `return fmt.Sprintf(ErrMaxChunkBytesHit, ql.maxChunkBytesPerQuery)` would remove dependency on `validation` package. Calling code (querier package) can add this wrapping when needed.

Reply:
I jump in on this. It's in the TODO list, but I suggested to do it in a follow up PR to keep changes easier to review.